### PR TITLE
Add metrics to Redis Sink Single

### PR DESCRIPTION
**Pull request by conorbros**

### CODE REVIEW

File: shotover-proxy/src/transforms/redis/sink_single.rs [hunk 1]
```diff
@@ -89,7 +98,10 @@ impl Transform for RedisSinkSingle {
 
         match outbound_framed_codec.next().fuse().await {
             Some(a) => a,
-            None => Err(anyhow!("couldnt get frame")),
+            None => {
+                counter!("redis_single_failed_request", 1, "chain" => self.chain_name.clone());
```
Comment (rukai) [inline shotover-proxy/src/transforms/redis/sink_single.rs:102]: This doesnt seem to match up with when the cluster sink emits this counter

Comment (conorbros) [reply shotover-proxy/src/transforms/redis/sink_single.rs]: Ah I see the cluster records specifically Redis error message not just a general error.

Comment (conorbros) [reply shotover-proxy/src/transforms/redis/sink_single.rs]: Aren't those "shotover generated redis errors" just actual Redis errors being passed back? 

We aren't changing the type of Redis error or anything about the message.

Comment (rukai) [reply shotover-proxy/src/transforms/redis/sink_single.rs]: ah, true!
So the functionality for the counter looks like: record every error the remote server sends back do not record any shotover side errors.

**Review by rukai — APPROVED** [review]
LGTM!

File: shotover-proxy/src/transforms/redis/sink_single.rs [hunk 2]
```diff
@@ -35,20 +39,27 @@ pub struct RedisSinkSingle {
...
+            self.tls.clone(),
+            self.chain_name.clone(),
+        )
     }
 }
 
 impl RedisSinkSingle {
-    pub fn new(address: String, tls: Option<TlsConnector>) -> RedisSinkSingle {
+    pub fn new(address: String, tls: Option<TlsConnector>, chain_name: String) -> RedisSinkSingle {
+        register_counter!("redis_single_failed_request", Unit::Count, "chain" => chain_name.clone());
```
Comment (benbromhead) [inline shotover-proxy/src/transforms/redis/sink_single.rs:57]: Naming wise we should call this "failed_requests" and add a tag "transform" => "redis_sink_single"

File: shotover-proxy/src/transforms/redis/sink_single.rs [hunk 3]
```diff
@@ -88,7 +99,16 @@ impl Transform for RedisSinkSingle {
             .ok();
 
         match outbound_framed_codec.next().fuse().await {
-            Some(a) => a,
+            Some(a) => {
+                if let Ok(ref messages) = a {
+                    for message in messages {
+                        if let RawFrame::Redis(Frame::Error(_)) = message.original {
+                            counter!("redis_single_failed_request", 1, "chain" => self.chain_name.clone());
```
Comment (benbromhead) [inline shotover-proxy/src/transforms/redis/sink_single.rs:106]: same here

**Review by benbromhead — CHANGES_REQUESTED** [review]
Also clear out the commented out test code (no longer applicable btw)

File: shotover-proxy/src/transforms/redis/sink_single.rs (hunk 2 above)
Comment (conorbros) [reply shotover-proxy/src/transforms/redis/sink_single.rs:57]: Should redis cluster also have these changes?

**Review by benbromhead — APPROVED** [review]

**Review by rukai — APPROVED** [review]
Hmmm I wonder if we should remove the use of `CONTEXT_CHAIN_NAME` from sink_cluster and just do it the same way as sink_single.
Regardless lets this PR as is, and potentially clean that up later.

**Pull request merged**